src/simpleNbox.cpp

```diff
@@ -113,9 +127,36 @@ void SimpleNbox::setData( const std::string &varName,
     
     std::string biome = SNBOX_DEFAULT_BIOME;
     std::string varNameParsed = varName;
+    std::vector<std::string>::const_iterator it_global;
+    it_global = std::find(biome_list.begin(), biome_list.end(), SNBOX_DEFAULT_BIOME);
+    
     if( splitvec.size() == 2 ) {    // i.e., in form <biome>.<varname>
         biome = splitvec[ 0 ];
         varNameParsed = splitvec[ 1 ];
+        if ( has_biome( SNBOX_DEFAULT_BIOME ) ) {
+            H_LOG( logger, Logger::DEBUG ) << "Removing biome '" << SNBOX_DEFAULT_BIOME <<
+                "' because you cannot have both 'global' and biome data. " << std::endl;
+            // We don't use the `deleteBiome` function here because
+            // when `setData` is used to initialize the core from the
+            // INI file, most of the time series variables that
+            // `deleteBiome` modifies have not been initialized yet.
+            // This should be relatively safe because (1) we check
+            // consistency of biome-specific variable sizes before
+            // running, and (2) the R interface will not let you use
+            // `setData` to modify the biome list.
+            biome_list.erase( it_global );
```

Comment:
This is currently failing on Travis because of a known bug in old versions of GCC/clang:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=57158
https://stackoverflow.com/questions/19559235/missing-const-iterator-overload-of-stdvectorerase-with-g-4-8/19559316

The root cause of this is probably that the JGCRI/Hector Travis instance is still using [Ubuntu Trusty](https://travis-ci.org/JGCRI/hector/jobs/551480463#L9); the same code built without issue on my fork because my Travis containers have been [upgraded to Ubuntu Xenial](https://travis-ci.org/ashiklom/hector/jobs/551467449#L9).

I'm perfectly happy to change this from a `const_iterator` to just a plain `iterator`, which should fix this specific issue, but I'm not totally clear on the consequences of that, so I defer to you all for advice.

Reply:
Is there a reason _not_ to change the Hector repo to use Xenial?

Reply:
> Is there a reason not to change the Hector repo to use Xenial?

If we want to support the older OS and compiler. GCC 4.8 is only ~5 years old, and is actually later than the default GCC on, say, PIC (4.4.7). There's no way I would have caught this issue if Travis had upgraded the shared repo to Xenial, but it could have been a nasty surprise for someone trying to install Hector on PIC.

@rplzzz @pralitp -- Do you have any thoughts on this? Should we just abandon support for GCC < 4.9 (I think that's where this was fixed, but I'm not 100% sure), or should I implement some of the casting hacks/workarounds described in the SO question above?

Reply:
GCC 4.8 may be only 5 years old, but its C++-11 support is a little spotty, which is a good reason not to support it.  We already have to use gcc 6 or later for GCAM anyway.

BTW, if you've been habitually using one compiler on a particular system and you switch to another, you probably need to reinstall all of your R packages (which probably means it's just a good time to upgrade to a new version of R).  Having packages built with a mix of different compilers can lead to weird, random crashes.

Reply:
Thanks! I explicitly set the Travis build environment distribution to Xenial in 974685f485b27f04313d49f1278f44fdf0354b58.